### DISCUSSION TITLE
Octane: how to choose number of workers?

### DISCUSSION THREAD

**Discussion by enzorevo** [laravel/framework#50059] [Q&A]
Hi, is there any best practice for choosing the number of workers for Laravel Octane (I use it with Roadrunner)? 

Thanks,

Lorenzo

**Comment by ghost** [comment]
Hi. Question is unclear. Are you talking about the queue?

**Comment by enzorevo** [comment]
Hi, I refer to the params inserted when doing octane:start, specifically —workers and —max-requests.

**Reply by mihaileu** [reply]
you should do some test with different values based on your current hardware. Choose the one that satisfy you performance needs.

**Comment by AlliBalliBaba** [comment]
I think you usually are doing fine with the default number of workers (1 worker per cpu kernel). If the worker number is too low you might not get the full performance out of all the CPU cores of your system. If the worker number is too high you will actually loose performance because the CPU has to do too much context switching between all the workers. From my experience the default or a little higher leads to similar results. Just be aware that it may be calculated too highly in some specific virtualized environments like WSL.
